addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3880,6 +3880,19 @@ export class OdooEditor extends EventTarget {
             if (!focusNode) {
                 return;
             }
+            // We are adding a zws before the code element and set selection next to it.
+            // Because on delete forward the zws before the code element will be removed and
+            // when we try to hit left arrow cursor won't get out of the code block.
+            const codeElement = closestElement(anchorNode, 'code');
+            if (
+                codeElement?.classList.contains('o_inline_code') &&
+                !anchorOffset &&
+                (!codeElement.previousSibling || codeElement?.previousSibling.nodeType !== Node.TEXT_NODE ) &&
+                !isZWS(codeElement?.previousSibling)
+            ) {
+                codeElement.before(document.createTextNode('\u200B'));
+                setSelection(codeElement.previousSibling, 1);
```

Comment:
I get setting the selection to offset 1 for the right arrow but equivalently, shouldn't the left one be at offset 0? Looking below if I'm not mistaken, a selection at zws 1 could be changed to the start of the next leaf.
I notice, testing your branch, that the only way to write before inline code at the start of its parent, is to use the Home key. Maybe this is why?

Reply:
Yes, This scenario skipped my mind. Thank you, I'll make the changes. :+1:

Reply:
Thanks. The problem remains though. Seeing that the code after yours is looking for a zws before the selection _in a variable_, maybe that's the issue. I'm thinking anyway you shouldn't get through this code after you've made your change so the rest of it should be in an `else` or something.

Reply:
I have added rest of its code in else condition.

Reply:
Great, now left behaves like right :-)